src/plugins/discover/public/application/angular/discover.js

```diff
@@ -370,25 +370,19 @@ function discoverController($route, $scope, Promise) {
     });
   };
 
-  $scope.searchSource
-    .setField('index', $scope.indexPattern)
-    .setField('highlightAll', true)
-    .setField('version', true);
-
-  // Even when searching rollups, we want to use the default strategy so that we get back a
-  // document-like response.
-  $scope.searchSource.setPreferredSearchStrategyId('default');
+  persistentSearchSource.setField('index', $scope.indexPattern);
 
   // searchSource which applies time range
-  const timeRangeSearchSource = savedSearch.searchSource.create();
+  const volatileSearchSource = savedSearch.searchSource.create();
```

Comment:
Should these changes be applied to embeddable as well? https://github.com/elastic/kibana/blob/c37b0e1474ba3dd7b8994048122e8dd8349b61e0/src/plugins/discover/public/application/embeddable/search_embeddable.ts#L211

Reply:
As mentioned in the description:

> The saved search embeddable is still polluting the original search source, but it doesn't matter here because it's never saved.

We can structure it the same way there, but it's not related to the issue of polluted saved objects because the embeddable search source is never saved. To keep the scope small I would like to do these things in 7.13

Reply:
Sorry, missed this line in the description